src/session_impl.cpp

```diff
@@ -6891,7 +6896,7 @@ namespace {
 		// connected to the internet. The whole point is to forward ports through
 		// the gateway
 		if ((s->flags & listen_socket_t::local_network)
-			|| (s->flags & listen_socket_t::proxy))
+			|| (s->flags & !m_settings.get_bool(settings_pack::listen_on_proxy)))
```

Comment:
This is not right. the `&` is a bit-wise AND operation. you can mask the `flags` field against flags/bits set in this field. `get_bool()` returns a `bool`, not a flag value. I think you meant to preserve the existing check and add this additional one.